Sources/Snapshot.swift

```diff
@@ -0,0 +1,38 @@
+/// Represents a snapshot of a collection.
+///
+/// The changeset associated with the first snapshot received by any given observation
+/// is generally irrelevant. Observations should special case the first snapshot as a
+/// complete replacement, or whatever semantic that fits their purpose.
+///
+/// The previous version of the collection is unavailable on the initial snapshot,
+/// since there is no history to refer to.
+public struct Snapshot<Collection: Swift.Collection>: SnapshotProtocol {
+	/// The previous version of the collection, or `nil` if `self` is an initial snapshot.
+	public let previous: Collection?
+
+	/// The current version of the collection.
+	public let current: Collection
+
+	/// The changeset which, when applied on `previous`, reproduces `current`.
+	public let changeset: Changeset
```

Comment:
(A little bit more context)

```swift
let receipts: MutableProperty<[Receipt]>([])
let diffedReceipts = receipts.diff()

diffedReceipts.producer.start() // See the `initial` event.

receipts.value = [Receipt(id: 1), Receipt(id: 2)]
diffedReceipts.producer.start() // Not sending out the `initial` event.
```

We could tackle this by (1) limiting collection diffing to _properties_; and (2) special implementation for the diffing result property, whose producer should always send `initial` initially.